src/pybind/mgr/snap_schedule/module.py

```diff
@@ -65,11 +68,15 @@ def snap_schedule_get(self,
         '''
         List current snapshot schedules
         '''
-        use_fs = fs if fs else self.default_fs
-        if not self.has_fs(use_fs):
-            return -errno.EINVAL, '', f"no such filesystem: {use_fs}"
+        if not fs:
+            if self.get_num_filesystems() > 1:
+                return -errno.EINVAL, '', "filesystem argument is required; there are more than one filesystem"
+            else:
+                fs = self.default_fs
```

Comment:
Since the only place `default_fs` is accessed is in these changes, I think just move this check to that `@property default_fs` method.

Maybe also change it to `_default_fs` to make it clear that it's a private member.

Reply:
* updated to validate `fs` in `_default_fs` property method
* added qa test class for testing mandatory --fs argument
* added doc commit with note
* added `PendingReleaseNotes` commit with note

[Teuthology run](https://pulpito.ceph.com/mchangir-2023-08-17_07:19:05-fs-wip-mchangir-mgr-snap_schedule-make-fs-argument-mandatory-distro-default-smithi/) with 5 of 6 passing jobs
1 job failed with:
```
cluster [WRN] Health check failed: 1 pool(s) do not have an application enabled (POOL_APP_NOT_ENABLED
```